### PR TITLE
Intl.NumberFormat

### PR DESCRIPTION
**Pull request by TorsteinHonsi**
Made the `lang.decimalPoint` and `lang.thousandsSep` default to, in prioritized order: the current `lang.locale` setting, page `html.lang` attribute, or the browser default language.

#### Upgrade note
Decimal points and thousands separators now default to the page or browser language setting. To stick to the previous defaults, set `lang.decimalPoint` to `.` and `lang.thousandsSep` to ` ` (a single space). You can also use the new `lang.locale` option to specify the preferred language for number and date formatting.

---

For now: Testing if it is viable to use Intl.NumberFormat for localized numbers similar to the Intl.DateFormat function.

#### Status 2024-11-5
There is not much to gain from rewriting how numbers are formatted, but we risk regressions, browser quirks and performance hits. In the #21934 PR, we take another approach by just replacing the default `decimalPoint` and `thousandsSep` with those of the current locale.

#### New conclusion 2024-11-08
The other approach didn't work out because it mixes non-latin separator into latin numerals, for example for languages like Farsi. Trying again in this PR.

 - [x] Test and check visuals against master
 - [x] Change base to v12-staging
 - [x] Use `chart.locale` instead of the temporary local access in `Templating.numberFormat`
 - [x] Benchmark. Consider caching for NumberFormat instances, similar to DateFormat.
 - [x] Update the [internationalization docs article](https://www.highcharts.com/docs/advanced-chart-features/internationalization). Move the three demos from jsFiddle to the `highcharts` repo.

### CODE REVIEW

File: ts/Core/Templating.ts [hunk 1]
```diff
@@ -422,86 +424,88 @@ function numberFormat(
...
 
-    if (+exponent[1] < 0 && !firstDecimals) {
-        ret = '0';
-    } else {
-        // Add the remaining thousands groups, joined by the thousands separator
-        ret += strinteger
-            .substr(thousands)
-            .replace(/(\d{3})(?=\d)/g, '$1' + thousandsSep);
+    /*
+    @todo After merging this into v12-staging, use chart.locale
```
Comment (pawelfus) [inline ts/Core/Templating.ts:480]: Is this still the case?

**Review by pawelfus — APPROVED** [review]
Looks good!

File: ts/Core/Templating.ts (hunk 1 above)
Comment (TorsteinHonsi) [reply ts/Core/Templating.ts:480]: No, it is already done.

File: ts/Core/Defaults.ts [hunk 2]
```diff
@@ -228,11 +230,12 @@ const defaultOptions: DefaultOptions = {
...
         /**
          * The default decimal point used in the `Highcharts.numberFormat`
-         * method unless otherwise specified in the function arguments.
+         * method unless otherwise specified in the function arguments. Defaults
+         * to the locale decimal point as determined by `lang.locale`.
          *
          * @since 1.2.2
          */
-        decimalPoint: '.',
+        decimalPoint: void 0,
```
[the remark this answers is not in the document]
Comment (TorsteinHonsi) [reply ts/Core/Defaults.ts:238]: Good point! I was unsure how that would work with the upcoming TS-based doclets, but we'll cross that bridge when we get there. 8 bytes saved ✔️

**Pull request merged**